src/Driver.cpp

```diff
@@ -28,10 +30,15 @@ void Driver::open(std::string const& uri)
     openURI(uri);
 }
 
-void Driver::read()
+bool Driver::read()
 {
-    readPacket(&buffer[0], buffer.size());
-    parseMessage(&buffer[4],buffer.size()-4);
+    int msg_size = readPacket(&buffer[0], buffer.size());
+    if(msg_size == KVH_1750_MSG_SIZE)
```

Comment:
> But what is the harm in checking return values?

It basically shows that you don't understand what contracts you need to have enforced in your codebase, a.k.a. that you don't understand the codebase ("not understanding" is fine, defending your lack of understanding is not). After a few hundred of "no harm in testing X" you basically make everyone's life a living hell - including yours a few months from now. As a direct example, here, you return a boolean on a method that can only return `true`. And then test for it in a component, creating two dead paths.

Moreover, it shows that the analysis you did for a bug (with the commit messages I've seen, I am assuming that you're trying to fix a bug) is wrong, since the "fix" most probably does not change anything.

Contracts (implicit or explicit ones) are the bread and butter of what makes large-scale software development viable. If you're afraid of having a bug in `extractPacket`, thoroughly unit test it.